k8s/operators/src/diskpool/crd.rs

```diff
@@ -67,22 +71,39 @@ pub enum PoolState {
 }
 
 #[derive(Debug, Clone, Serialize, Deserialize, Eq, PartialEq, JsonSchema)]
-/// Status of the pool which is driven and changed by the controller loop
+#[non_exhaustive]
+/// PoolStatus is Control plane status of a given DSP CR.
+pub enum PoolStatus {
```

Comment:
PoolState sounds a lot like PoolStatus. IMHO it needs a more descriptive name to distinguish it from PoolState, so that it is obvious to developers what it means.

Reply:
This is mirroring the control-plane where State is composed of a Status and other fields, eg: `VolumeState { status: Online, replicas: 3, .... }`
But it becomes messy here with the PoolState above which I think we might be able to rename to `CrdStatus` instead?

Reply:
Im okay with CrStatus.